### PR TITLE
Incoming webhook doc updates

### PR DESCRIPTION
**Pull request by jasonblais**
Updated:

Submitting a pull request for incoming webhook doc updates.

Not sure how I can decrease the image sizes though, what I tried doesn't seem to work. They're larger in size than necessary.

### CODE REVIEW

File: source/developer/slash-commands.rst [hunk 1]
```diff
@@ -3,6 +3,46 @@
...
+.. toctree::
+	:maxdepth: 2
+
+Creating a simple slash command
+-----------------------------------
+
+Parameters and formatting
+--------------------------
+
+Subheading
```
Comment (lfbrock) [inline source/developer/slash-commands.rst]: Is this supposed to say subheading? Or is it placeholder/WIP?

Comment (jasonblais) [inline source/developer/slash-commands.rst]: The slash commands and outgoing webhooks docs are WIP.

Just the incoming webhooks is ready for review for content and structure

Comment (jasonblais) [inline source/developer/slash-commands.rst]: @lfbrock I've undone the changes on those two docs to reduce confusion.

File: source/developer/webhooks-incoming.rst [hunk 2]
```diff
@@ -3,128 +3,213 @@
...
+Messages with advanced formatting can be created by including an :doc:`attachment array <message-attachments>` in the JSON payload.
 
-4. With **Enable integrations to override profile picture icons** turned on, you can similarly change the icon the message posts with by providing a link to an image in the *icon_url* parameter of your payload. For example, ``payload={"icon_url": "http://example.com/somecoolimage.jpg", "text": "Hello, this is some text."}`` will post using whatever image is located at *http://example.com/somecoolimage.jpg* as the icon for the post.
+Complete incoming webhook example
```
Comment (lfbrock) [inline source/developer/webhooks-incoming.rst]: Wondering if we should show a complete example at the top so information is available at a glance

File: source/developer/webhooks-incoming.rst [hunk 3]
```diff
@@ -3,128 +3,213 @@
...
+
+4. The external application may be written in any programming language as long as it supports sending an HTTP POST request in the required JSON format to a specified Mattermost URL.
+
+5. For the HTTP request body, if `Content-Type` is specified as `application/json` in the header of the HTTP request, then the body can be direct JSON. For example,
+
+  .. code-block::
+
+    {"text": "Hello, this is some text."}
+
+6. It is often best to set up your integration running on Heroku, an AWS server or a server of your own before to test sending messages to Mattermost channels.
```
Comment (lfbrock) [inline source/developer/webhooks-incoming.rst]: I think this sentence might be missing some words, not quite sure what it means

**Comment by jasonblais** [comment]
Should be ready for review.

Not sure how I can decrease the image sizes though, what I tried doesn't seem to work. They're larger in size than necessary.

File: source/developer/webhooks-incoming.rst [hunk 4]
```diff
@@ -3,132 +3,210 @@
...
+To send a message to a direct message channel, add an "@" symbol followed by the username to the ``channel`` parameter.
+
+  .. code-block::
+
+    payload={"channel": "@username", "text": "Hello, this is some text\nThis is more text. :tada:"}
+
+Override the username
+~~~~~~~~~~~~~~~~~~~~~
+
+You can override the username the messages posts as by specifying a ``username`` parameter in your JSON payload.
```
Comment (ccbrown) [inline source/developer/webhooks-incoming.rst]: Subject-verb agreement error: "messages posts"

**Comment by jasonblais** [comment]
Thanks @ccbrown! Fixed it

**Pull request merged**